tests/Feature/Http/Controllers/PostControllerTest.php

```diff
@@ -2,27 +2,167 @@
 
 namespace Tests\Feature\Http\Controllers;
 
-use App\Models\Category;
+use Tests\TestCase;
+use App\Models\Post;
 use App\Models\User;
-use Illuminate\Foundation\Testing\RefreshDatabase;
+use App\Models\Category;
 use Illuminate\Foundation\Testing\WithFaker;
-use Tests\TestCase;
+use Illuminate\Foundation\Testing\RefreshDatabase;
 
 class PostControllerTest extends TestCase
 {
     use RefreshDatabase;
     use WithFaker;
 
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->user = User::factory()->create()->assignRole('user');
+        $this->moderator = User::factory()->create()->assignRole('moderator');
+        $this->category = Category::factory()->create();
+        $this->post = new Post([
+            'title' => 'Test',
+            'content' => 'Test',
+        ]);
+        $this->post->category()->associate($this->category);
+        $this->user->posts()->save($this->post);
+
+        $this->post->category()->associate($this->category);
+        $this->user->posts()->save($this->post);
+
+    }
+
     /**
      * @test
      */
     public function it_should_return_the_post_create_view()
     {
-        $user = User::factory()->create()->assignRole('user');
-        $category = Category::factory()->create();
+        $response = $this->actingAs($this->user)->get(route('posts.create', $this->category));
+        $response->assertViewIs('posts.create');
+    }
 
-        $response = $this->actingAs($user)->get(route('posts.create', $category));
+    /**
+     * @test
+     */
+    public function a_moderator_can_delete_any_posts()
+    {
 
-        $response->assertViewIs('posts.create');
+        $this->assertDatabaseHas('posts', [
+            'title' => 'Test',
+            'content' => 'Test',
+            'category_id' => $this->category->id,
+            'user_id' => $this->user->id,
+            'deleted_at' => null
+        ]);
+        
+        $this->actingAs($this->moderator);
+        $this->delete(route('posts.destroy', $this->post));
+        $this->assertSoftDeleted('posts', [
+            'id' => $this->post->id,
+            'title' => 'Test',
```

Comment:
I would check only for the 'id' here, you are not interested in the title, content, category_id, ...